Compatibility: test_streaming.TestStreamingRestartToAnotherVersion do_write_read returns empty list on stable-only restarts/mixed clusters

## Summary

`TestStreamingMixedCluster::test_mixed_cluster` and `TestStreamingRestartToAnotherVersion::test_restart_to_another_version` fail with the consumer reading an empty list (`assert [] == ['{"error_count":...', '{"error_count":...']`) — the produced records never reach the reader. The configurations affected here **do not necessarily include a `prestable-26-3` daemon** (e.g. `mixed_stable-26-2-1`, `restart_stable-26-2-1_to_stable-26-2-1`, `restart_stable-26-2-1_to_stable-26-1-1`), which distinguishes them from [#40261](https://github.com/ydb-platform/ydb/issues/40261) (closed). No daemon crash accompanies the assertion — it fails cleanly.

Tracker: [#40898](https://github.com/ydb-platform/ydb/issues/40898).

## Environment

- **Workflow run:** https://github.com/ydb-platform/ydb/actions/runs/26185317746
- **Job:** `Regression-run_compatibility (relwithdebinfo) / main:relwithdebinfo`
- **Build preset:** `relwithdebinfo`
- **Branch / build commit:** `main` @ [`06e00a9c6159ab2cf14c614b5073d4c128fa2fbd`](https://github.com/ydb-platform/ydb/commit/06e00a9c6159ab2cf14c614b5073d4c128fa2fbd)
- **Versions under test:**
  - init  — `stable-26-1-1` @ [`d41184c58e51d579b645a97621ef4f2c58eeb8c4`](https://github.com/ydb-platform/ydb/commit/d41184c58e51d579b645a97621ef4f2c58eeb8c4)
  - inter — `stable-26-2-1` @ [`b115abba219ba5f5f3abcf318c068902a5ed08a8`](https://github.com/ydb-platform/ydb/commit/b115abba219ba5f5f3abcf318c068902a5ed08a8)
  - target — `prestable-26-3` @ [`875bf20ceec86926802064a89bb5a04133d5bb0e`](https://github.com/ydb-platform/ydb/commit/875bf20ceec86926802064a89bb5a04133d5bb0e)

## HTML reports

- [try_1](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/26185317746/ya-main-x86-64/try_1/ya-test.html#FAIL)
- [try_2](https://storage.yandexcloud.net/ydb-gh-logs/ydb-platform/ydb/Regression-run_compatibility/26185317746/ya-main-x86-64/try_2/ya-test.html#FAIL)

## Affected tests

| Test | try_1 | try_2 |
|------|:-----:|:-----:|
| `ydb/tests/compatibility/streaming/test_streaming.py::TestStreamingMixedCluster::test_mixed_cluster[mixed_stable-26-2-1]` | FAIL | FAIL |
| `ydb/tests/compatibility/streaming/test_streaming.py::TestStreamingMixedCluster::test_mixed_cluster[mixed_stable-26-2-1_and_stable-26-1-1]` | FAIL | n/a (not retried) |
| `ydb/tests/compatibility/streaming/test_streaming.py::TestStreamingRestartToAnotherVersion::test_restart_to_another_version[restart_prestable-26-3_to_prestable-26-3]` | FAIL | FAIL |
| `ydb/tests/compatibility/streaming/test_streaming.py::TestStreamingRestartToAnotherVersion::test_restart_to_another_version[restart_stable-26-2-1_to_prestable-26-3]` | FAIL | FAIL |
| `ydb/tests/compatibility/streaming/test_streaming.py::TestStreamingRestartToAnotherVersion::test_restart_to_another_version[restart_stable-26-2-1_to_stable-26-1-1]` | FAIL | FAIL |
| `ydb/tests/compatibility/streaming/test_streaming.py::TestStreamingRestartToAnotherVersion::test_restart_to_another_version[restart_stable-26-2-1_to_stable-26-2-1]` | FAIL | FAIL |

## Error

```
ydb/tests/compatibility/streaming/test_streaming.py:188: in test_restart_to_another_version
    self.do_test_part1()
ydb/tests/compatibility/streaming/test_streaming.py:151: in do_test_part1
    self.do_write_read(input, expected_data)
ydb/tests/compatibility/streaming/test_streaming.py:139: in do_write_read
    assert sorted(read_data) == sorted(expected_output)
E   assert [] == ['{"error_count":1,"host":"host-2","ts":"2025-01-01T00:00:00Z"}',
                  '{"error_count":2,"host":"host-1","ts":"2025-01-01T00:00:00Z"}']
E     Right contains 2 more items, first extra item: '{"error_count":1,"host":"host-2","ts":"2025-01-01T00:00:00Z"}'
E     Full diff:
E     + []
E     - [
E     -     '{"error_count":1,"host":"host-2","ts":"2025-01-01T00:00:00Z"}',
E     -     '{"error_count":2,"host":"host-1","ts":"2025-01-01T00:00:00Z"}',
E     - ]
```

No accompanying daemon crash on these — see the per-test stderr; the failure is a pure assertion mismatch after the rolling/mixed/restart cycle.

## Reproduction

```bash
./ya make -tA --build relwithdebinfo \
  -DYDB_COMPAT_INIT_REF=stable-26-1-1 \
  -DYDB_COMPAT_INTER_REF=stable-26-2-1 \
  -DYDB_COMPAT_TARGET_REF=prestable-26-3 \
  ydb/tests/compatibility/streaming -F "*restart_stable-26-2-1_to_stable-26-2-1*"
```

## Related (closed)

- [#40261](https://github.com/ydb-platform/ydb/issues/40261) — same `do_write_read` empty-read assertion but only on `prestable-26-3` configurations, accompanied by SIGSEGV.
- [#33778](https://github.com/ydb-platform/ydb/issues/33778) — same assertion on older version pair.

<!-- nemesis:related-versions:start -->

---

### Related issues

- Also filed (stable-26-2-1-enterprise): #50416

<!-- nemesis:related-versions:end -->

--- a/ydb/tests/compatibility/streaming/test_streaming.py
+++ b/ydb/tests/compatibility/streaming/test_streaming.py
@@ -80,7 +80,7 @@ class StreamingTestBase:
                     SELECT host, COUNT(*) AS error_count, CAST(HOP_START() AS String) AS ts
                     FROM $filtered
                     GROUP BY
-                        HoppingWindow(CAST(time AS Timestamp), 'PT600S', 'PT600S'),
+                        HOP(CAST(time AS Timestamp), 'PT600S', 'PT600S', 'PT0S'),
                         host
                 );
 

